numpy/random/mtrand/mtrand.pyx

```diff
@@ -4871,7 +4871,7 @@ cdef class RandomState:
         if isinstance(x, (int, long, np.integer)):
             arr = np.arange(x)
         else:
-            arr = np.array(x)
+            arr = x
```

Comment:
Hang on, this means that `x` is now modified in place, rather an a copy. This isn't ok.